### DISCUSSION TITLE
Error when trying to connect via Bluetooth using RBPi

### DISCUSSION THREAD

**Discussion by jctsoccer12** [iRobotEducation/create3_docs#453] [Questions]
### How are you connecting to your Create 3?

Bluetooth (Python SDK)

### Computer(s) Model(s) and Operating System(s)

Raspberry Pi 4 / Raspbian 11

### Which version of ROS 2 is installed on your computer?

None

### Which firmware version is installed on your robot?

G.5.3

### Which RMW is your robot running?

CycloneDDS

### Are there multiple Create 3 robots connected to your network?

None

### Is multicast enabled?

None

### What is the Adapter Board's USB/BLE Toggle currently switched to?

Bluetooth (default)

### Describe your question.

I'm attempting to use the Python SDK with a Raspberry Pi 4 to allow my Create3 to track light using a BH1750 sensor. When I attempt to run example code from the SDK library, I'm getting a connection error that looks like this:
`raise BleakDeviceNotFoundError(
bleak.exc.BleakDeviceNotFoundError: Device with address 00:16:A4:D2:22:54 was not found. It may have been removed from BlueZ when scanning stopped.`

I did notice that when I run the code, the first output in the shell is `Connecting to iRobot-67665C4AA76C4E648753E3 (00:16:A4:D2:22:54)`, but when I check the robot's bluetooth name via wifi, it is different than that. Is there a place I need to go to change this setting somewhere in the code?

I'm able to connect to the robot just fine using my Mac and the Python Web Playground, but not having any luck from the Pi. Any tips or tricks to fix this error would be sincerely appreciated - I'm super new to working with this sort of thing! I'm happy to provide more information if I'm missing anything.

**Comment by shamlian** [comment]
Hello!

Easy question first: the Bluetooth name, if set via the webserver, is not shown on advertisement until the application is restarted in the webserver (if it is set over Bluetooth, it is changed immediately). On your next app restart or reboot, the name should show up correctly, unless your OS has cached it.

Harder question: why isn't the robot connecting in Linux? It could be a few things. One thing to check is your version of BlueZ; there was a bug in BlueZ that made the LE connections to the robot flaky which has been recently solved. You can get your version by typing `bluetoothd -v`. I am not exactly sure what version it was fixed in (I believe 5.64; I am currently running 5.66). Another thing to check is the version of the `irobot-edu-sdk`; versions before 0.4.0 had a bug which caused disconnection problems (use `pip show irobot-edu-sdk` or `pip install irobot-edu-sdk --upgrade` to just make sure you have the latest).

The first thing you'll have to do is remove your robot's MAC address from the BlueZ cache if it's still there; the directions for that can be found at https://github.com/iRobotEducation/irobot-edu-python-sdk/issues/3 .

At this point, see if you can successfully connect. If not, you will likely need to upgrade BlueZ. The directions for doing that can be found at https://github.com/iRobotEducation/irobot-edu-python-sdk/issues/24 . Let us know if you need any further help or clarification.

**Reply by jctsoccer12** [reply]
Thank you so much for the reply! My sincere apologies for the delayed response - been trying lots of different things, and thankfully I've got it working now. The solution might be a bit unsatisfying: a fresh install of Raspbian ended up fixing things. I really appreciate the support!

**Reply by shamlian** [reply]
Glad you've got it resolved but it's too bad that the root cause seems to not be clear. Hopefully all stays well. Have fun!

**Reply by shamlian** [reply]
(I've closed this for now, feel free to reopen if something changes.)
